### PR TITLE
Command line switch to specify language on startup (#10044) (taken over after abandoned #10089)

### LINKED ISSUE #10044: Command line switch to specify language on startup
#9648 # Is your feature request related to a problem? Please describe.
I was flying blind (with no speech) in the preferences dialog, attempting to direct speech output to an audio device that'd just been plugged in when I accidentally set my language to Hebrew. Admittedly this was user error, but once set, getting the language switched back to my usual choice of "User default" was a task and a half.

### Describe the solution you'd like
Could a command line switch to force language on startup be implemented? Section 15.1 of the user guide has all sorts of useful switches, but I didn't see language mentioned there.

### Describe alternatives you've considered
Well, I guess I could be more careful... :)

### PR DESCRIPTION
**Pull request by lukaszgo1**
Many thanks for @JulienCochuyt  for laying the foundation of this work!

### Link to issue number:
Fixes #10044
Supersedes #10089

### Summary of the issue:
It can sometimes be useful to specify NVDA's language from the CLI. One use case which has been mentioned in #10044 is when language has been accidentally changed to a foreign one. This can also be useful for developers (it would certainly be a welcome addition for me when working on #12250 and #12753).

### Description of how this pull request fixes the issue:
Add a new `--lang` command line parameter.
It accepts either `"Windows"` or the usual `"en"` / `"de_CH"` codes, with a little tolerance regarding to case and dashes.
As discussed during the review in #10089 the parameter is treated as a temporary override (i.e. it does not affect configuration at all and is effective only until NVDA is restarted).
As a bonus this PR adds unit tests for `languageHandler.normalizeLanguage` and `languageHandler.getAvailableLanguages` (tests for the former were necessary to confirm it provides us with normalization we need, and since I had to split the latter one into two functions added tests made it possible to confirm that current behavior has not changed).

### Testing strategy:
- Started NVDA without languate specified from the CLI - ensured that language can be changed.
- Started NVDA with valid language code from the CLI - ensured that specified language has indeed been applied, that settings dialog shows "Command line: lang_code" as the selected language, that the language is set to the one specified from the configuration after restart and that language can still be changed from the settings dialog.
- Started NVDA with invalid language - made sure that error dialog is shown and that the help message contains all supported language codes.
### Known issues with pull request:
- Resetting NVDA configuration does not change language properly (this is not new for this PR).
- In #10089 @leonardder suggested  to use `globalVars.appArgs.language` to store the current language the same way we do for config path regardless if it has been provided from the command line or not.  While I like this idea myself it causes installer failure, and while I'd like to look into this further this should be done as a follow up.
### Change log entries:
New features
- The new `--lang` command line parameter allows to override the configured NVDA language.

### Code Review Checklist:

- [X] Pull Request description:
  - description is up to date
  - change log entries
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] API is compatible with existing add-ons.
- [X] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

**Comment by CyrilleB79** [comment]
I have not tested yet; I hope to do it soon.

But I think that  `--lang=Windows` should be kept because it is not in the list. And if I am not mistaken, it's not the same thing to provide `--lang=en` and `--lang=Windows` even if your Windows is in English.

**Comment by michaelDCurran** [comment]
in nvda.pyw, the given language argument is checked to see if it is in 
the list produced by languageHandler.listNVDALocales.

This pr also seems to insert 'windows' into that list in the 
listNVDALocales function, thus it is already an acceptable value.

**Review by michaelDCurran — APPROVED** [review]

**Comment by lukaszgo1** [comment]
@Qchristensen  Would it be possible for you to review the user guide change? I'd like to have this merged sooner than later since as mentioned in the known issues of this PR there is a follow up work I'd like to do, but since it would not be backwards compatible it has to be done during development cycle for a .1 release. Also @CyrilleB79  are you happy with the latest changes?

**Comment by CyrilleB79** [comment]
Hi Lukasz

I have not yet had the opportunity to test the last build. But from the description you have made of the lasts commits, I am fully happy with it. Many thanks!

**Comment by CyrilleB79** [comment]
And I have now been able to download and test the snapshot containing your last modifications. All is fine for me. Thanks again!

Hope to have @Qchristensen's approval soon to see this PR merged.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -8,6 +8,7 @@ What's New in NVDA
 == New Features ==
 - Support for reporting notes in MS Excel with UI Automation enabled on Windows 11. (#12861)
 - In recent builds of Microsoft Word via UI Automation on Windows 11, the existence of bookmarks, draft comments and resolved comments are now reported in both speech and braille. (#12861)
+- The new --lang command line parameter allows to override the configured NVDA language. (#10044)
```
Comment (Qchristensen) [inline user_docs/en/changes.t2t:11]: The meaning is there but this line reads awkwardly.  I'd suggest either:
- The new --lang command line parameter allows you to override the configured NVDA language. (#10044) 
 or
- The new --lang command line parameter allows overriding the configured NVDA language. (#10044)

**Review by Qchristensen — CHANGES_REQUESTED** [review]
User Guide text is fine, but I'd like a small change to the changes.t2t as proposed.  In any case, great work on a useful feature!

File: user_docs/en/changes.t2t (hunk 1 above)
Comment (lukaszgo1) [reply user_docs/en/changes.t2t:11]: @michaelDCurran  Given external contributors aren't really supposed to touch change log file would you be willing to modify the changes entry as per @Qchristensen's sugggestion when reviewing the latest commits and merging this PR? Aside from this it seems this is ready to go to master.

**Comment by michaelDCurran** [comment]
Yes, I'll handle it from here.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**